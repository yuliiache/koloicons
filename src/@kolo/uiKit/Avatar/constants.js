export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
