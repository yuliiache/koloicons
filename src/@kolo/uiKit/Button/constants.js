export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
