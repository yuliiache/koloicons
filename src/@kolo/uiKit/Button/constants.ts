export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
