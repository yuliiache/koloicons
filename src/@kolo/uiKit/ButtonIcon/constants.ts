export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
