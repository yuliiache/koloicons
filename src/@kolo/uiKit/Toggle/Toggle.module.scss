@import "@kolo/design";

.toggle {
  display: flex;
  align-items: center;

  font-size: 0.7rem;
  line-height: 1.43;
  color: $secFontColor;

  &Input {
    @include vh;

    &:focus + .toggleLabel {
      background-color: $IconColor;
      box-shadow: 0 0 0 4px rgba($BrandColor, 0.2);
    }

    &:checked:disabled + .toggleLabel {
      background-color: #b8d0ff;

      &:hover {
        background-color: #b8d0ff;
      }

      &::before {
        background-color: $WhiteBG;
        opacity: 0.5;
      }
    }

    &:checked + .toggleLabel {
      background-color: $BrandColor;

      &:hover {
        background-color: #659fff;
      }

      &::before {
        left: 22px;
      }
    }

    &:disabled {
      & + .toggleLabel {
        background-color: #e4e8f2;
        cursor: auto;

        &:hover {
          background-color: #e4e8f2;
        }

        &::before {
          background-color: rgba(175, 181, 196, 0.8);
        }
      }
    }
  }

  &Label {
    position: relative;
    margin-right: 8px;
    display: inline-block;
    flex-shrink: 0;
    width: 40px;
    height: 20px;
    box-sizing: border-box;

    background-color: $InterfaceIcon;
    border: none;
    border-radius: 10px;
    cursor: pointer;
    transition: background-color 0.3s ease-in;

    &:hover {
      background-color: $IconColor;
    }

    &::before {
      content: "";
      position: absolute;
      top: 2px;
      left: 2px;

      width: 16px;
      height: 16px;

      background-color: $WhiteBG;
      border-radius: 50%;

      transition: left 0.3s;
    }
  }
}
