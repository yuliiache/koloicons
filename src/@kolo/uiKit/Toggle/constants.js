export const TOGGLE_TEST_ID = 'toggleComponent';
