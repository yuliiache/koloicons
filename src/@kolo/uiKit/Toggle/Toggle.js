import React from 'react';
import PropTypes from 'prop-types';

import {TOGGLE_TEST_ID} from './constants';

import styles from './Toggle.module.scss';

export const Toggle = ({isChecked = false, isDisabled = false, name, compId = 'toggleId', onChange}) => {
  return (
    <label
      className={styles.toggle}
      htmlFor={compId}
    >
      <input
        className={styles.toggleInput}
        type="checkbox"
        name={name}
        id={compId}
        data-testid={TOGGLE_TEST_ID}
        checked={isChecked}
        disabled={isDisabled}
        onChange={onChange}
      />
      <span className={styles.toggleLabel} />
    </label>
  );
};

Toggle.propTypes = {
  isChecked: PropTypes.bool,
  isDisabled: PropTypes.bool,
  name: PropTypes.string.isRequired,
  compId: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};
