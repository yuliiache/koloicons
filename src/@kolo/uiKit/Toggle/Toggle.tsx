import {FC} from 'react';

import {TOGGLE_TEST_ID} from './constants';
import styles from './Toggle.module.scss';

interface Props {
  name: string;
  compId?: string;
  isChecked?: boolean;
  isDisabled?: boolean;
  onChange: () => void;
}

export const Toggle: FC<Props> = ({name, compId = 'toggleId', isChecked = false, isDisabled = false, onChange}) => {
  return (
    <label
      className={styles.toggle}
      htmlFor={compId}
    >
      <input
        className={styles.toggleInput}
        type="checkbox"
        name={name}
        id={compId}
        data-testid={TOGGLE_TEST_ID}
        checked={isChecked}
        disabled={isDisabled}
        onChange={onChange}
      />
      <span className={styles.toggleLabel} />
    </label>
  );
};
