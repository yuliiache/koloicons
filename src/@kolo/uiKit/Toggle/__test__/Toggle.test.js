import React from 'react';
import {cleanup, render, screen, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {TOGGLE_TEST_ID} from '../constants';
import {Toggle} from '../Toggle';

describe('Toggle', () => {
  afterEach(() => cleanup());

  it('should render snapshot successfuly', () => {
    const props = {
      name: 'toggle',
      onChange: jest.fn(),
    };

    const {asFragment} = render(<Toggle {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have "checked" attribute as false when toggle is off', () => {
    const props = {
      isChecked: false,
      name: 'toggle',
      onChange: jest.fn(),
    };

    render(<Toggle {...props} />);

    const toggle = screen.getByTestId(TOGGLE_TEST_ID);

    expect(toggle.checked).toBe(false);
  });

  it('should have "checked" attribute as true when toggle is on', () => {
    const props = {
      isChecked: true,
      name: 'toggle',
      onChange: jest.fn(),
    };

    render(<Toggle {...props} />);

    const toggle = screen.getByTestId(TOGGLE_TEST_ID);

    expect(toggle.checked).toBe(true);
  });

  it('should have "disabled" attribute as false when toggle is not disabled', () => {
    const props = {
      isDisabled: false,
      name: 'toggle',
      onChange: jest.fn(),
    };

    render(<Toggle {...props} />);

    const toggle = screen.getByTestId(TOGGLE_TEST_ID);

    expect(toggle).not.toHaveAttribute('disabled');
  });

  it('should have "disabled" attribute as true when toggle is disabled', () => {
    const props = {
      isDisabled: true,
      name: 'toggle',
      onChange: jest.fn(),
    };

    render(<Toggle {...props} />);

    const toggle = screen.getByTestId(TOGGLE_TEST_ID);

    expect(toggle).toHaveAttribute('disabled');
  });

  it('should call onChange function from props when toggle was switched', async () => {
    const onChange = jest.fn();
    const props = {
      name: 'toggle',
      onChange,
    };

    render(<Toggle {...props} />);

    const toggle = screen.getByTestId(TOGGLE_TEST_ID);
    userEvent.click(toggle);

    await waitFor(() => expect(onChange).toHaveBeenCalled());
  });
});
