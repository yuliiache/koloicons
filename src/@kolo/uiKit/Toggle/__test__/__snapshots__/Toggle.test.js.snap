// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render snapshot successfuly 1`] = `
<DocumentFragment>
  <label
    class="toggle"
    for="toggleId"
  >
    <input
      class="toggleInput"
      data-testid="toggleComponent"
      id="toggleId"
      name="toggle"
      type="checkbox"
    />
    <span
      class="toggleLabel"
    />
  </label>
</DocumentFragment>
`;
